src/js/select.js

```diff
@@ -227,22 +228,29 @@ Menu.prototype._createMenuEl = function(wrapperEl, selectEl) {
       // add row item to menu
       rowEl = doc.createElement('div');
       rowEl.textContent = loopEl.textContent;
-      rowEl._muiIndex = indexNum;
-
-      // handle selected options
-      if (loopEl.selected) {
-        rowEl.className = selectedClass;
-        selectedRow = menuEl.children.length;
-      }
 
       // handle optgroup options
       if (inGroup) jqLite.addClass(rowEl, 'mui-optgroup__option');
 
-      menuEl.appendChild(rowEl);
+      if (loopEl.disabled) {
+        // do not attach muiIndex to disable <option> elements to make them
+        // unselectable.
+        rowEl.className = disabledClass;
+      } else {
+        rowEl._muiIndex = indexNum;
+
+        // handle selected options
+        if (loopEl.selected) {
+          rowEl.className = selectedClass;
+          selectedRow = menuEl.children.length;
+        }
+      }
 
       // add to index map
       indexMap[indexNum] = rowEl;
```

Comment:
Yes, my suggestion would require a deeper change in the code to convert `indexMap` to an Array and `currentIndex` to an array position tracker. I'm already very grateful for your contribution so I don't mind merging your pull request now and doing the refactor separately.

Let me know your preference.